### PR TITLE
Internally use estimate_gas for settlement simulation at current block

### PR DESCRIPTION
**Pull request by vkgnosis**
Estimate gas internally simulates the transaction which tells us whether
the simulation succeeded and how much gas was used in one call.

Next I will change the driver to use this function so that we do not simulate twice.

### Test Plan
Adapted the ignored test to call the new function and ran the test.

### CODE REVIEW

**Comment by vkgnosis** [comment]
Draft while I debug an issue.

File: solver/src/settlement_simulation.rs [hunk 1]
```diff
@@ -85,18 +90,27 @@ pub async fn simulate_settlements(
...
-                    }
-                    Block::LatestWithoutTenderly => err,
-                }
+                Error::new(err).context(tenderly_link(block, network_id, transaction_builder))
             })
         })
         .collect())
 }
 
+fn method(
```
Comment (josojo) [inline solver/src/settlement_simulation.rs:99]: This function name seems quite unspecific, maybe get_settle_method() would be a better name?

**Review by josojo — APPROVED** [review]
Nice, LGTM

File: solver/src/driver.rs [hunk 2]
```diff
@@ -230,8 +231,8 @@ impl Driver {
             .into_iter()
             .zip(simulations)
             .partition_map(|(settlement, result)| match result {
-                Ok(()) => Either::Left(settlement),
-                Err(err) => Either::Right((settlement, err)),
+                Ok(_) => Either::Left(settlement),
```
Comment (nlordell) [inline solver/src/driver.rs:234]: One advantage of `Ok(())` is that if the type of the `result` changes, you would get a compiler error to update at this location. Very minor though, and I also see the argument that `Ok(_)` could more clearly communicate the intent of "it succeeded, don't care about the result".

Not requesting any changes, just rambling.

File: solver/src/driver.rs [hunk 3]
```diff
@@ -246,26 +247,29 @@ impl Driver {
...
-                tracing::error!(
-                    "unable to complete simulation of settlements at earlier block {}: {:?}",
-                    current_block_during_liquidity_fetch,
-                    err
-                );
-                return;
-            }
-        };
+        let simulations =
+            match settlement_simulation::simulate_at_custom_block_and_error_with_tenderly_link(
```
Comment (nlordell) [inline solver/src/driver.rs:251]: naming nit:
```suggestion
            match settlement_simulation::simulate_and_error_with_tenderly_link(
```

**Review by nlordell — APPROVED** [review]
:pray: Looks good!

Approving assuming stray `dbg!(_)` will get cleaned up before merging.

File: solver/src/driver.rs (hunk 2 above)
Comment (vkgnosis) [reply solver/src/driver.rs:234]: This is exactly why it's _ now. The type has changed from () to U256 for the gas estimation. Next PR will make use of the value.

File: solver/src/settlement_simulation.rs [hunk 4]
```diff
@@ -124,30 +138,41 @@ mod tests {
...
             &contract,
             &web3,
+            0.0,
             network_id.as_str(),
-            Block::FixedWithTenderly(block),
+            block,
+        )
+        .await
+        .unwrap();
+        let _ = dbg!(result);
```
[the remark this answers is not in the document]
Comment (nlordell) [reply solver/src/settlement_simulation.rs:166]: Oh this is test code :man_facepalming:

**Pull request merged**